components/table/nz-table.component.ts

```diff
@@ -307,4 +333,15 @@ export class NzTableComponent<T = any> implements OnInit, AfterViewInit, OnDestr
     this.destroy$.next();
     this.destroy$.complete();
   }
+
+  private handleDataChange(nzData: NzDataSource<T> | T[]): void {
+    this.data = isDataSource(nzData) ? nzData : new NzArrayDataSource(nzData, !this.nzFrontPagination);
```

Comment:
nzData maybe null or undefined

Reply:
@vthinkxie Fixed, if `null | undefined`, an empty array will be used.